src/orange/g4org/SolidConverter.cc

```diff
@@ -0,0 +1,640 @@
+{
+    double vol = this->calc_capacity(solid_base);
+    double radius = std::cbrt(vol / (4.0 / 3.0 * constants::pi));
+    return make_shape<Sphere>(solid_base, radius);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Convert a solid that's not in the cache.
+ */
+auto SolidConverter::convert_impl(arg_type solid_base) -> result_type
+{
+    using ConvertFuncPtr = result_type (SolidConverter::*)(arg_type);
+    using MapTypeConverter
+        = std::unordered_map<std::type_index, ConvertFuncPtr>;
+
+    // clang-format off
+    #define SC_TYPE_FUNC(MIXED, LOWER) \
+    {std::type_index(typeid(G4##MIXED)), &SolidConverter::LOWER}
+    static const MapTypeConverter type_to_converter = {
+        SC_TYPE_FUNC(Box              , box),
+        SC_TYPE_FUNC(Cons             , cons),
+        SC_TYPE_FUNC(CutTubs          , cuttubs),
+        SC_TYPE_FUNC(DisplacedSolid   , displaced),
+        SC_TYPE_FUNC(Ellipsoid        , ellipsoid),
+        SC_TYPE_FUNC(EllipticalCone   , ellipticalcone),
+        SC_TYPE_FUNC(EllipticalTube   , ellipticaltube),
+        SC_TYPE_FUNC(ExtrudedSolid    , extrudedsolid),
+        SC_TYPE_FUNC(GenericPolycone  , genericpolycone),
+        SC_TYPE_FUNC(GenericTrap      , generictrap),
+        SC_TYPE_FUNC(Hype             , hype),
+        SC_TYPE_FUNC(IntersectionSolid, intersectionsolid),
+        SC_TYPE_FUNC(Orb              , orb),
+        SC_TYPE_FUNC(Para             , para),
+        SC_TYPE_FUNC(Paraboloid       , paraboloid),
+        SC_TYPE_FUNC(Polycone         , polycone),
+        SC_TYPE_FUNC(Polyhedra        , polyhedra),
+        SC_TYPE_FUNC(ReflectedSolid   , reflectedsolid),
+        SC_TYPE_FUNC(Sphere           , sphere),
+        SC_TYPE_FUNC(SubtractionSolid , subtractionsolid),
+        SC_TYPE_FUNC(TessellatedSolid , tessellatedsolid),
+        SC_TYPE_FUNC(Tet              , tet),
+        SC_TYPE_FUNC(Torus            , torus),
+        SC_TYPE_FUNC(Trap             , trap),
+        SC_TYPE_FUNC(Trd              , trd),
+        SC_TYPE_FUNC(Tubs             , tubs),
+        SC_TYPE_FUNC(UnionSolid       , unionsolid),
+    };
+    // clang-format on
+#undef SC_TYPE_FUNC
+
+    // Look up converter function based on the solid's C++ type
+    auto func_iter
+        = type_to_converter.find(std::type_index(typeid(solid_base)));
+
+    result_type result = nullptr;
+    CELER_VALIDATE(func_iter != type_to_converter.end(),
+                   << "unsupported solid type "
+                   << TypeDemangler<G4VSolid>{}(solid_base));
+
+    // Call our corresponding member function to convert the solid
+    ConvertFuncPtr fp = func_iter->second;
+    result = (this->*fp)(solid_base);
+    CELER_ENSURE(result);
+    return result;
+}
+
+//---------------------------------------------------------------------------//
+// CONVERTERS
+//---------------------------------------------------------------------------//
+//! Convert a box
```

Comment:
Since this is an actual doxygen comment, shouldn't it end in a period?

Reply:
The one-line "briefs" like this one don't need it... I decided to use those instead of full doc blocks since the functions are generally short and obvious